src/ActionPreviewHeader.jsx

```diff
@@ -1,44 +1,55 @@
-import React from 'react';
+import React, { Component } from 'react';
 
-const ActionPreviewHeader = ({
-  styling, inspectedPath, onInspectPath, tab, onSelectTab
-}) => {
-  return (
-    <div key='previewHeader' {...styling('previewHeader')}>
-      <div {...styling('tabSelector')}>
-        {['Action', 'Diff', 'State'].map(t =>
-          <div onClick={() => onSelectTab(t)}
-               key={t}
-               {...styling([
-                 'selectorButton',
-                 t === tab && 'selectorButtonSelected'
-               ], t === tab)}>
-            {t}
-          </div>
-        )}
-      </div>
-      <div {...styling('inspectedPath')}>
-        {inspectedPath.length ?
-          <span {...styling('inspectedPathKey')}>
-            <a onClick={() => onInspectPath([])}
-               {...styling('inspectedPathKeyLink')}>
-              {tab}
-            </a>
-          </span> : tab
-        }
-        {inspectedPath.map((key, idx) =>
-          idx === inspectedPath.length - 1 ? key :
-          <span key={key}
-             {...styling('inspectedPathKey')}>
-            <a onClick={() => onInspectPath(inspectedPath.slice(0, idx + 1))}
-               {...styling('inspectedPathKeyLink')}>
-              {key}
-            </a>
-          </span>
-        )}
+class ActionPreviewHeader extends Component {
+  constructor(props) {
+    super(props);
+    this.tabs = ['Action', 'Diff', 'State'];
```

Comment:
Well, it's not really stateless if it stores something in component :)

As for arrow functions - it's not a high-order component that contains some heavy components inside, so I didn't bother much about it. The problem usually is not that we create new function, but that we have new instance of `onClick` prop each time, so child component re-renders every time. In this case it doesn't affect much as this component contains just a few DOM elements.

(what could actually help with performance is to add `shouldPureComponentUpdate` - for some reason I missed it in other components)